token_review.go

```diff
@@ -189,7 +187,7 @@ func mockTokenReviewFactory(name, namespace, UID string) tokenReviewFactory {
 	}
 }
 
-func (t *mockTokenReview) Review(ctx context.Context, cjwt string, aud []string) (*tokenReviewResult, error) {
+func (t *mockTokenReview) Review(ctx context.Context, cjwt string, aud []string, client *http.Client) (*tokenReviewResult, error) {
```

Comment:
I think the suggestion elsewhere was essentially that you could make this mock function return an error if the client is not configured correctly. i.e. something like:

```go
httpTransport, ok := client.Transport.(*http.Transport)
if !ok {
	return nil, errors.New("failed to check whether DisableKeepAlives is false as Transport is not *http.Transport")
}
if httpTransport.DisableKeepAlives {
	return nil, errors.New("expected DisableKeepAlives to be false but was true")
}
```

An alternative might be to add a similar check more directly on the returned value from `Backend()` in a new test.

Reply:
Updated in https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/142/commits/246c3f85023b0bcfd46ebc3f667658b274e74cf5